t/e2e/jail/jail_rpc_test.go

```diff
@@ -191,6 +215,30 @@ func (s *JailRPCTestSuite) TestContractDeployment() {
 
 	expectedResponse := txHash.Hex()
 	s.Equal(expectedResponse, response)
+
+	if testContractMining {
+		s.waitForBoolValue(cell, "wasMined", true, 3*time.Minute, "timeout while waiting for the contract to be mined")
+	}
+}
+
+func (s *JailRPCTestSuite) waitForBoolValue(cell jail.JSCell, name string, expected bool, timeout time.Duration, msg string) {
+	deadline := time.Now().Add(timeout)
+
+	s.T().Logf("waiting for the variable '%s' to be '%v' with timeout of '%v'", name, expected, timeout)
+
+	for time.Now().Before(deadline) {
```

Comment:
Any reason not to use a for loop and ticker from `time` package?

Reply:
no, not really.

Reply:
I can tweak it later, if needed, but I don't think it will bring any clarity benefits here :)